daemon/client.go

```diff
@@ -26,38 +27,50 @@ type Client interface {
 }
 
 type client struct {
-	proctordHost string
-	emailId      string
-	accessToken  string
+	proctordHost          string
+	emailId               string
+	accessToken           string
+	connectionTimeoutSecs time.Duration
 }
 
 type ProcToExecute struct {
 	Name string            `json:"name"`
 	Args map[string]string `json:"args"`
 }
 
-func NewClient() Client {
+func NewClient(proctorConfig config.ProctorConfig) Client {
 	return &client{
-		proctordHost: config.ProctorHost(),
-		emailId:      config.EmailId(),
-		accessToken:  config.AccessToken(),
+		proctordHost:          proctorConfig.Host,
```

Comment:
Can we make proctorConfig an interface? When Host/Email/AccessToken are available to modify, mistakes can happen. Instead we can make them package private and have getters `Host(), Email(), AccessToken()`, etc.

Reply:
Also making it an interface helps us mocking values in test at a later stage. (Instead of doing the complicated setup we do currently with the test config file)